p2p-sync/src/main/java/org/smartregister/p2p/view/QRCodeScannerView.java

```diff
@@ -0,0 +1,204 @@
+                .setFacing(CameraSource.CAMERA_FACING_BACK)
+                .setRequestedPreviewSize(1600, 1024)
+                .setAutoFocusEnabled(true)
+                .setRequestedFps(45.0f);
+
+        cameraSource = builder.build();
+    }
+
+    @Override
+    public void release() {
+        //Todo
+    }
+
+    @Override
+    public void receiveDetections(Detector.Detections<Barcode> detections) {
+        final SparseArray<Barcode> barcodeSparseArray = detections.getDetectedItems();
+        if (barcodeSparseArray.size() > 0) {
+            Vibrator vibrator = (Vibrator) getContext().getApplicationContext().getSystemService(Context.VIBRATOR_SERVICE);
+            assert vibrator != null;
+            vibrator.vibrate(100);
+
+            for (OnQRRecognisedListener onQRRecognisedListener : onQRRecognisedListeners) {
+                onQRRecognisedListener.onBarcodeRecognised(barcodeSparseArray);
+            }
+        }
+    }
+
+
+    /**
+     * Starts or restarts the camera source, if it exists.  If the camera source doesn't exist yet
+     * (e.g., because onResume was called before the camera source was created), this will be called
+     * again when the camera source is created.
+     */
+    public void startCameraSource() throws SecurityException {
+        int code = GoogleApiAvailability.getInstance().isGooglePlayServicesAvailable(
+                getContext().getApplicationContext());
+        if (code != ConnectionResult.SUCCESS) {
+
+            if (getContext() instanceof Activity) {
+                Dialog errorDialog =
+                        GoogleApiAvailability.getInstance().getErrorDialog((Activity) getContext()
+                                , code, Constants.RQ_CODE.BARCODE_SCANNER_GOOGLE_PLAY_FIX);
+                errorDialog.show();
+            } else {
+                Timber.e("Could not show Google Play Services resolution dialog since it was not called from an activity");
+            }
+        }
+
+        if (cameraSource != null) {
+            try {
+                cameraSourcePreview.start(cameraSource);
+            } catch (IOException e) {
+                Timber.e(e, "Unable to start camera source.");
+                cameraSource.release();
+                cameraSource = null;
+            }
+        }
+    }
+
+    /**
+     * Restarts the camera.
+     */
+    @OnLifecycleEvent(Lifecycle.Event.ON_RESUME)
+    public void onResume() {
+        startCameraSource();
+    }
+
+    /**
+     * Stops the camera.
+     */
+    @OnLifecycleEvent(Lifecycle.Event.ON_PAUSE)
+    public void onPause() {
+        if (cameraSourcePreview != null) {
+            cameraSourcePreview.stop();
+        }
+    }
+
+    /**
+     * Releases the resources associated with the camera source, the associated detectors, and the
+     * rest of the processing pipeline.
+     */
+    @OnLifecycleEvent(Lifecycle.Event.ON_DESTROY)
+    public void onDestroy() {
+        if (cameraSourcePreview != null) {
```

Comment:
Also release the `cameraSource`

Reply:
The `cameraSource` when attached to the `CameraSourcePreview` is released in the `CameraSourcePreview` if the `cameraSource` was attached succesfully, otherwise, it is released in the block where the attaching to the `CameraSourcePreview` failed